draft-ietf-dnsop-svcb-httpssvc.md

```diff
@@ -743,38 +745,49 @@ responses to the address queries that were issued in parallel.
 
-## "alpn"
-
-The "alpn" SvcParamKey defines the Application Layer Protocol
-(ALPN, as defined in {{!RFC7301}) supported by a TLS-based alternative
-service.  Its value SHOULD be an entry in the IANA registry "TLS
-Application-Layer Protocol Negotiation (ALPN) Protocol IDs".
-
-The presentation format and wire format of SvcParamValue
-is its registered "Identification Sequence".  This key SHALL NOT
-appear more than once in a SvcFieldValue.
-
-Clients MUST include this value in the ProtocolNameList in their
-ClientHello's `application_layer_protocol_negotiation` extension.
-Clients SHOULD also include any other values that they support and
-could negotiate on that connection with equivalent or better security
-properties.  For example, when using a SvcFieldValue with an "alpn" of
-"h2", the client MAY also include "http/1.1" in the ProtocolNameList.
-
-Clients MUST ignore SVCB RRs where the "alpn" SvcParamValue
-is unknown or not supported for use with the current scheme.
-
-The value of the "alpn" SvcParamKey can have effects beyond the content
-of the TLS handshake and stream.  For example, an "alpn" value of "h3"
-({{HTTP3}} Section 11.1) indicates the client must use QUIC, not TLS.
+## "transport" and "no-default-transport"
+
+The "transport" and "no-default-transport" SvcParamKeys together
+indicate the set of transport protocols supported by this service endpoint.
+
+Each scheme that is mapped to SVCB defines a set or registry of allowed
+transport values, and a "default set" of supported values, which SHOULD NOT
+be empty.  To determine the set of transport protocols supported by an
+endpoint (the "transport set"), the client collects the set of
+"transport" values, and then adds the default set unless the
+"no-default-transport" SvcParamKey is present.  The presence of a value in
+the transport set indicates that this service endpoint, described by
+SvcDomainName and the other parameters (e.g. "port") offers service with
+that transport.
+
+Clients SHOULD NOT attempt connection to a service endpoint whose
+transport set does not contain any compatible transport protocols.  To ensure
+consistency of behavior, clients MAY reject the entire SVCB RRSet and fall
+back to basic connection establishment if all of the RRs indicate
+"no-default-transport", even if connection could have succeeded using a
+non-default transport.
+
+For "transport", the presentation and wire format are the same,
+in the expectation that values will typically be ASCII strings, but any
+sequence of octets is a permissible value.  This key MAY appear multiple times
+with different values.
+
+The value of "no-default-transport" MUST be empty, and clients MUST reject
+the RR if a nonempty value is encountered.  This key may appear at most once
+in an RR.  For compatibility with clients that require default transports,
+zone operators SHOULD ensure that at least one RR in each RRSet supports the
+default transports.
 
```

Comment:
Perhaps add:
"The wire format for the value of no-default-transport is empty (a SvcParamValue with a length of zero octets)."
or something else to be crystal clear that value is still present in the wire format but empty.

Reply:
OK, I added clarifying text to this effect.